app/master/cluster_master.py

```diff
@@ -62,12 +65,24 @@ def api_representation(self):
             'slaves': slaves_representation,
         }
 
-    def builds(self):
+    def builds(self, offset: int=None, limit: int=None) -> List['Build']:
         """
-        Returns a list of all builds
-        :rtype: list[Build]
+        Returns a list of all builds. Both offset and limit are assumed to be nonnegative at this point
+        since the corrections or negative offsets and limits are done when the query parameters are first initialized.
```

Comment:
I'm fine with leaving negative offsets for later. @ethomas2 I think in your example it should be `offset=-1&limit=1`. I think it only makes sense to support negative offsets. Negative limits (especially combined with negative offsets) feels unintuitive to me.